src/oop/programs_with_interesting_classes/easy/q4/MainTest.java

```diff
@@ -0,0 +1,258 @@
+        return Stream.of(
+                Arguments.of("Oak", 21.54, "Apple", 54.3),
+                Arguments.of("Apple", 1.43, "Pistachio", 324.4),
+                Arguments.of("Orange", 0.5, "White", 32.3),
+                Arguments.of("White", 35.43, "Brown", 75.4),
+                Arguments.of("Brown", 36.23, "Coconut", 34.1),
+                Arguments.of("Nut", 364.32, "Oak", 92.3),
+                Arguments.of("Pistachio", 36.24, "Orange", 83.8),
+                Arguments.of("Coconut", 232.432, "Nut", 3.2)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void treeConstructorInitializesValuesCorrectly(String value1, double value2) throws Throwable {
+        String wrongValueMessage = "The %s constructor did not initialize the %s attribute to the correct value based on the parameters passed to the constructor.";
+        Object[][] constructorArgs = {
+                {value1, String.class},
+                {value2, double.class}
+        };
+        Object checkupInstance = testClass.createInstance(constructorArgs);
+        _assertEquals(value1, testClass.getFieldValue(checkupInstance, attributeName1), String.format(wrongValueMessage, className, attributeName1));
+        _assertEquals(value2, testClass.getFieldValue(checkupInstance, attributeName2), String.format(wrongValueMessage, className, attributeName2));
+    }
+
+    @Test
+    public void gettingOlderIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(methodName, null), String.format(incorrectMethodDefinition, methodName, className));
+        assertTrue(testClass.hasModifier(methodName, null, "public"), String.format(incorrectModifierMessage, methodName, className));
+        assertTrue(testClass.hasReturnType(methodName, null, Void.TYPE), String.format(incorrectReturnType, methodName, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void treeClassHasCorrectGettingOlderMethod(String value1, double value2) throws Throwable {
+        Object[][] constructorArgs = {
+                {value1, String.class},
+                {value2, double.class}
+        };
+        Object classInstance = testClass.createInstance(constructorArgs);
+        String incorrectNumDays = "The %s method in the %s class does not correctly change the %s attribute.";
+        testClass.callMethod(methodName, classInstance);
+        _assertEquals(value2 + 4, testClass.getFieldValue(classInstance, attributeName2),
+                String.format(incorrectNumDays, methodName, className, attributeName2));
+    }
```

Comment:
As this is a parsons question, I think we can just combine these two tests, so you can check the method call signature and then check that the method works.

Reply:
And the same for the other methods.